pkg/pillar/attest/attest.go

```diff
@@ -0,0 +1,346 @@
+		triggerSelfEvent(ctx, EventAttestEscrowRecorded)
+		return nil
+	}
+	fmt.Printf("Error %v while sending attest escrow keys\n", err)
+	switch err {
+	case ErrControllerUnavailable:
+		return startNewRetryTimer(ctx)
+	case ErrInfoTokenInvalid:
+		ctx.state = StateRestartWait
+		startNewRetryTimer(ctx)
+	default:
+		return fmt.Errorf("Unknown error %v", err)
+	}
+	//keep the compiler happy
+	return nil
+}
+
+func handleAttestEscrowRecordedWhlAttestEscrowWait(ctx *Context) error {
+	fmt.Println("handleAttestEscrowRecordedWhlAttestEscrowWait")
+	ctx.state = StateComplete
+	if ctx.restartRequestPending {
+		ctx.state = StateRestartWait
+		startNewRetryTimer(ctx)
+	}
+	return nil
+}
+
+func handleRestartWhlStateComplete(ctx *Context) error {
+	fmt.Println("handleRestartWhlStateComplete")
+	ctx.state = StateRestartWait
+	return startNewRetryTimer(ctx)
+}
+
+func handleRestartWhlStateInProgress(ctx *Context) error {
+	fmt.Println("handleRestartWhlStateInProgress")
+	ctx.restartRequestPending = true
+	return nil
+}
+
+func handleNonceMismatchWhlAttestWait(ctx *Context) error {
+	fmt.Println("handleNonceMismatchWhlAttestWait")
+	ctx.state = StateRestartWait
+	return startNewRetryTimer(ctx)
+}
+
+func handleQuoteMismatchWhlAttestWait(ctx *Context) error {
+	fmt.Println("handleQuoteMismatchWhlAttestWait")
+	return handleNonceMismatchWhlAttestWait(ctx)
+}
+
+func handleNoQuoteCertRcvdWhlAttestWait(ctx *Context) error {
+	fmt.Println("handleNoQuoteCertRcvdWhlAttestWait")
+	return handleNonceMismatchWhlAttestWait(ctx)
+}
+
+func handleAttestEscrowFailedWhlAttestEscrowWait(ctx *Context) error {
+	fmt.Println("handleAttestEscrowFailedWhlAttestEscrowWait")
+	ctx.state = StateRestartWait
+	return startNewRetryTimer(ctx)
+}
+
+func handleRetryTimerExpiryWhlRestartWait(ctx *Context) error {
+	fmt.Println("handleRetryTimerExpiryWhlRestartWait")
+	ctx.state = StateNone
+	return triggerSelfEvent(ctx, EventInitialize)
+}
+
+func handleRetryTimerExpiryWhlNonceWait(ctx *Context) error {
+	fmt.Println("handleRetryTimerExpiryWhlNonceWait")
+	ctx.state = StateNone
+	return triggerSelfEvent(ctx, EventInitialize)
+}
+
+func handleRetryTimerExpiryWhlAttestWait(ctx *Context) error {
+	//try re-sending quote
+	fmt.Println("handleRetryTimerExpiryWhlAttestWait")
+	return handleInternalQuoteRecvdWhlInternalQuoteWait(ctx)
+}
+
+func handleRetryTimerExpiryWhileAttestEscrowWait(ctx *Context) error {
+	//try re-sending escrow info
+	fmt.Println("handleRetryTimerExpiryWhileAttestEscrowWait")
+	return handleAttestSuccessfulWhlAttestWait(ctx)
+}
+
+func handleRetryTimerExpiryWhlInternalQuoteWait(ctx *Context) error {
+	fmt.Println("handleRetryTimerExpiryWhlInternalQuoteWait")
+	return handleNonceRecvdWhlNonceWait(ctx)
+}
+
+func despatchEvent(event Event, state State, ctx *Context) error {
+	elem, ok := transitions[Transition{event: event, state: state}]
+	if ok {
+		return elem(ctx)
+	} else {
+		return fmt.Errorf("Unexpected Event %s in State %s\n",
+			event.String(), state.String())
```

Comment:
Should this be a fatal? Or should there be a default handler (based on the event or state?)

Reply:
> Should this be a fatal? Or should there be a default handler (based on the event or state?)

Yes, it should be. Earlier I didn't use "log" package. Now, that is imported, and changed all fmt.Print[f|ln]'s to either log.[Debug|Error|Fatal]. With that, this above line has been changed to log.Fatalf

Reply:
OK